internal/cmd/run.go

```diff
@@ -43,34 +49,62 @@ func runCmd() *cobra.Command {
 				return errors.New("must provide at least one command to run")
 			}
 
-			runBlocks := make([]*document.CodeBlock, 0)
+			proj, err := getProject()
+			if err != nil {
+				return err
+			}
+
+			runBlocks := make([]project.FileCodeBlock, 0)
 
 			{
-				blocks, err := getCodeBlocks()
+				blocks, err := proj.LoadTasks()
 				if err != nil {
 					return err
 				}
 
 				if runMany {
-					for _, block := range blocks {
+					for _, fileBlock := range blocks {
+						block := fileBlock.Block
+
 						if !block.ExcludeFromRunAll() {
 							if category != "" && category != block.Category() {
 								continue
 							}
-							runBlocks = append(runBlocks, block)
+							runBlocks = append(runBlocks, fileBlock)
 						}
 					}
 					if len(runBlocks) == 0 {
 						return errors.New("No tasks to execute with the category provided")
 					}
 				} else {
 					for _, arg := range args {
-						block, err := lookupCodeBlock(blocks, arg)
+						queryFile, queryName, err := splitRunArgument(arg)
+						if err != nil {
+							return err
+						}
+
+						blocks, err := blocks.LookupWithFile(queryFile, queryName)
 						if err != nil {
 							return err
 						}
 
-						if err := replace(replaceScripts, block.Lines()); err != nil {
+						var block project.CodeBlock
+
+						if len(blocks) > 1 {
+							if !isTerminal(os.Stdout.Fd()) {
+								return errors.New("multiple matches found for code block; please use a file specifier in the form \"{file}:{task-name}\"")
```

Comment:
Here is the line for reference @christian-bromann

Reply:
Awesome 👍